src/types/figure.ts

```diff
@@ -1,14 +1,24 @@
-import { Pixel, UID } from ".";
+import { DOMCoordinates, Pixel, PixelPosition, Position, UID } from ".";
 
 export interface Figure {
   id: UID;
-  x: Pixel;
-  y: Pixel;
+  anchor: Position;
+  offset: PixelPosition;
+  fixed_position: boolean;
   width: Pixel;
   height: Pixel;
   tag: string;
 }
 
+export interface FigureUI extends DOMCoordinates {
+  figure: Figure;
+}
```

Comment:
We use a lot `figureUI.figure.id`, this makes the code harder to read IMHO.
Also, I'm not sure about the naming of `FigureUI`
What do you think of:
```ts
export interface FigureWithDomCoordinates extends Figure {
  domCoordinates: DOMCoordinates;
}
```